packages/powersync/lib/src/streaming_sync.dart

```diff
@@ -386,3 +393,63 @@ String? stringOrFirst(Object? details) {
     return null;
   }
 }
+
+/// Attempt to give a basic summary of the error for cases where the full error
```

Comment:
should we not move all of this into it's own file (including classes below)?

Reply:
I'm keeping this method here since it's private and only used in this one place, but I moved the classes to a separate exported file now.